# 小学期数据可视化作业

### 浙江的生育率怎么了？

###### 组员：颜秀炫

In [71]:
from pyecharts import options as opts
from pyecharts.charts import Line
from pyecharts.charts import Pie
from pyecharts.globals import ThemeType
import pandas as pd

In [72]:
# 导入输出图片工具
from pyecharts.render import make_snapshot
# 使用snapshot-selenium 渲染图片
from snapshot_selenium import snapshot

# 生育率曲线（左上角）

### 通过2010年与2020年生育率的对比，观察如今浙江省的生育率情况。

In [73]:
x_data=['2010','2020']
y_data=[1.22,1.01]
c=(
    Line(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .set_global_opts(
        tooltip_opts=opts.TooltipOpts(is_show=False),
        xaxis_opts=opts.AxisOpts(type_="category"),
        yaxis_opts=opts.AxisOpts(
            type_="value",
            axistick_opts=opts.AxisTickOpts(is_show=True),
            splitline_opts=opts.SplitLineOpts(is_show=True),
            min_=0.95, 
            max_=1.25,
        ),
    )
    .add_xaxis(xaxis_data=x_data)
    .add_yaxis(
        series_name="总和生育率",
        y_axis=y_data,
        symbol="emptyCircle",
        is_symbol_show=True,
        label_opts=opts.LabelOpts(is_show=False),
    )
   # .render("basic_line_chart.html")
)
# make_snapshot(snapshot, c.render(), "生育率02.png")
c.load_javascript()
c.render_notebook()

### 通过对比生育率曲线可以看出浙江省生育率的下跌，引起我们的思考：‘为什么会下跌？’，‘为什么下跌得如此迅速’，‘这背后有原因’，‘这是否是在警惕我们’

In [74]:
from pyecharts.charts import Pie

### 通过各种专家的研究与报告，大致可以从以下4个方面入手：

#### 1、结婚生育意愿


#### 2、受教育程度

#### 3、育龄群体

#### 4、经济压力

# 1、婚育意愿

### 生育的第一步首先是结婚

## 初婚数

In [76]:
c = (
    Bar(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .add_xaxis(['2014','2015','2016','2017','2018','2019'])
    .add_yaxis("初婚数（人）",[732704,648666,605781,572689,561895,435469] )
    .reversal_axis()
    .set_series_opts(label_opts=opts.LabelOpts(is_show=False))
    .set_global_opts(
        xaxis_opts=opts.AxisOpts(
            min_=100000, 
            max_=750000
        )
    )
    #.set_global_opts(title_opts=opts.TitleOpts(title="Bar-翻转 XY 轴"))
    #.render("bar_reversal_axis.html")
)
#make_snapshot(snapshot, c.render(), "初婚数.png")
c.load_javascript()
c.render_notebook()

### 由上图可知，浙江省每年初婚人数一直走低。这也从侧面意味着每年的生育预备种子选手一直在减少。

## 结婚率

In [77]:
x_data=['2010','2017','2018','2019','2020']
y_data1=[9.3, 6.1, 5.9, 5, 4.3]
c=(
    Line(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .set_global_opts(
        tooltip_opts=opts.TooltipOpts(is_show=False),
        xaxis_opts=opts.AxisOpts(type_="category"),
        yaxis_opts=opts.AxisOpts(
            type_="value",
            axistick_opts=opts.AxisTickOpts(is_show=True),
            splitline_opts=opts.SplitLineOpts(is_show=True),
            min_=4, 
            max_=10,
        ),
    )
    .add_xaxis(xaxis_data=x_data)
    .add_yaxis(
        series_name="结婚率",
        y_axis=y_data1,
        symbol="emptyCircle",
        is_symbol_show=True,
        is_smooth=True,
        label_opts=opts.LabelOpts(is_show=False),
        markpoint_opts=opts.MarkPointOpts(
            data=[opts.MarkPointItem(name="自定义标记点", coord=[x_data[4], y_data1[4]], value=y_data1[4])]
        ),
    )
   # .render("basic_line_chart.html")
)
# make_snapshot(snapshot, c.render(), "结婚率.png")
c.load_javascript()
c.render_notebook()

### 根据数据与报道，浙江省结婚率已经连续5年全国倒数第二，从直观上看，人们结婚意愿低。结婚率如此低，生育率又怎么会高呢？

### 结婚生育意愿低，或许可以追溯到第一条受教育程度中，受教育程度提高，特别是女性意识的觉醒，女性地位的提高。许多女性不再认为结婚生子是人生的必经阶段，男性也同样如此。

### 但我认为结婚生育意愿低的主要原因是经济压力大，买房买车嫁妆彩礼，都是人们结婚前需要考虑的问题。纵使浙江是全国城乡发展差距最小、城乡居民收入最高的省份，结婚成本仍是阻碍人们结婚的首要难关。

# 2、受教育情况

In [78]:
culture = pd.read_excel("受教育情况.xlsx")
culture

受教育程度       三普       四普       五普       六普       七普
0              大学    18.21    48.50   146.79   507.78  1097.03
1              高中   202.19   290.37   495.36   738.12   939.76
2              初中   691.55   983.98  1531.94  1996.41  2111.73
3              小学  1531.44  1643.92  1683.34  1568.54  1703.57
4          文盲、半文盲   930.67   723.64   321.85   306.10   175.44

In [79]:
c = (
    Pie(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .add(
        "",
        [list(z) for z in zip(culture.受教育程度, culture.七普)],
        radius=["30%", "55%"],
        center=["40%", "50%"]
    )
    .add(
        "",
        [list(z) for z in zip(culture.受教育程度, culture.三普)],
        center=["81%", "65%"],
        radius=["10%","20%"]
    )
    .set_global_opts(
        title_opts=opts.TitleOpts(title="受教育程度"),
        legend_opts=opts.LegendOpts(orient="vertical", pos_top="15%", pos_left="75%"),
    )
    #.set_series_opts(label_opts=opts.LabelOpts(formatter="{b}: {c}"))
    #.render("pie_radius.html")
)
c.load_javascript()
c.render_notebook()
#make_snapshot(snapshot, c.render(), "受教育程度02.png")

### 通过三普与七普的数据可以知道浙江省的受教育程度大幅提高，较直观的如大学与文盲半文盲的人数占比成反比

In [58]:
x_data=['三普','四普','五普','六普','七普']
y_data1=[0.54, 1.31, 3.51, 9.92,18.2]
y_data2=[27.58, 19.61, 7.7, 5.98, 2.91]
c=(
    Line(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .set_global_opts(
        tooltip_opts=opts.TooltipOpts(is_show=False),
        xaxis_opts=opts.AxisOpts(type_="category"),
        yaxis_opts=opts.AxisOpts(
            type_="value",
            axistick_opts=opts.AxisTickOpts(is_show=True),
            splitline_opts=opts.SplitLineOpts(is_show=True),
#             min_=0.95, 
#             max_=1.25,
        ),
    )
    .add_xaxis(xaxis_data=x_data)
    .add_yaxis(
        series_name="大学",
        y_axis=y_data1,
        symbol="emptyCircle",
        is_symbol_show=True,
        is_smooth=True,
        label_opts=opts.LabelOpts(is_show=False),
        markpoint_opts=opts.MarkPointOpts(
            data=[opts.MarkPointItem(name="自定义标记点", coord=[x_data[4], y_data1[4]], value=y_data1[4])]
        ),
    )
    .add_yaxis(
        series_name="文盲、半文盲",
        y_axis=y_data2,
        symbol="emptyCircle",
        is_symbol_show=True,
        is_smooth=True,
        label_opts=opts.LabelOpts(is_show=False),
        markpoint_opts=opts.MarkPointOpts(
            data=[opts.MarkPointItem(name="自定义标记点", coord=[x_data[0], y_data2[0]], value=y_data2[0])]
        ),
    )
   # .render("basic_line_chart.html")
)
#make_snapshot(snapshot, c.render(), "受教育折线02.png")
c.load_javascript()
c.render_notebook()

### 教育对人口数量的调控作用其中有一条是：教育具有控制人口增长的作用。即人口的平均文化越高，生育率越低。

### 所以：浙江人口整体受教育程度提高，相对的，生育率越低。

# 3、育龄群体

### 根据浙江省卫生健康委员会的文章，建国初至2010年以前浙江省一共有过两次婴儿潮。而1983年计划生育政策正式实施，这是否对浙江省育龄妇女人数有影响呢？

In [80]:
data_people = pd.read_excel(r'2-3 人口自然变动情况(1978-2019年).xlsx')
data_people.head(10)

年份   出生人数    出生率   死亡人数   死亡率   增长人数  自然增长率
0  1978  67.75  18.17  21.75  5.83  46.00  12.34
1  1979  67.82  17.98  22.23  5.89  45.59  12.09
2  1980  59.40  15.59  23.97  6.29  35.43   9.30
3  1981  69.00  17.93  24.12  6.27  44.89  11.66
4  1982  71.38  18.31  23.17  5.94  48.21  12.37
5  1983  62.66  15.89  25.13  6.37  37.53   9.52
6  1984  49.80  12.52  23.82  5.99  25.97   6.53
7  1985  50.59  12.61  24.25  6.05  26.34   6.56
8  1986  64.64  15.96  24.06  5.94  40.58  10.02
9  1987  69.67  17.01  28.34  6.92  41.33  10.09

In [81]:
c =(
    Line(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .set_global_opts(
        tooltip_opts=opts.TooltipOpts(is_show=False),
        xaxis_opts=opts.AxisOpts(type_="category"),
        yaxis_opts=opts.AxisOpts(
            type_="value",
            axistick_opts=opts.AxisTickOpts(is_show=True),
            splitline_opts=opts.SplitLineOpts(is_show=True),
            min_=40,
            max_=80
            #interval
        #datazoom_opts=opts.DataZoomOpts(orient="vertical")
        )

    )
    .add_xaxis(xaxis_data=data_people.年份)
    .add_yaxis(
        series_name="浙江省出生人数",
        y_axis=data_people.出生人数,
        symbol="emptyCircle",
        is_symbol_show=True,
        label_opts=opts.LabelOpts(is_show=False),
    )
)
c.load_javascript()
c.render_notebook()
# make_snapshot(snapshot, c.render(), "浙江省出生人数.png")

### 图中数据为1978年~2020年浙江省的出生人数。由于没有寻找到1978年以前的出生人数数据，无法看出第一代婴儿潮的年代，但根据卫生健康委员会的报告显示，第一代婴儿潮出现在60~70年代，第二代婴儿潮出现在80~90年代。

### 虽然由于1983年计划生育政策，使1984年与1985年的出生人数骤降，但由于第一代婴儿潮的影响，出现了第二代婴儿潮，于1988年达到顶峰。结束于1993年。往后十年浙江省出生人数一直处于下降趋势，偶有起伏，但影响不大。

### 2005年出生人数开始上升，至2014年，曲线虽起起伏伏但整体平缓。在这期间，除了第二代婴儿潮的影响，还有外来人口的影响。根据人口普查数据，2000年的省外人口为368.89万人，而2020年的省外人口为1619万人。浙江省成为外来人口第二多的省份。

### 2014年后浙江省出生人数开始上升。其中我认为有政策影响。2013年浙江省成为全国首个实施“单独二孩”政策，2016年全国开放“全面二胎”政策。有专家认为开放二胎可以扭转浙江省少子化，老龄化的趋势，但是，出生人数于2018年达到顶峰，随后下降，二胎堆积效应逐渐消失。

### 所以：在计划生育期间，育龄群体规模缩小，生育率随之降低。开放二胎后，出生人口数出现短暂提升，但对生育率影响不大。

### 浙江省老龄化也不容忽视

In [61]:
data = pd.read_excel('year.xlsx')
data.年龄

0     0-14岁
1    15-59岁
2    60岁及以上
Name: 年龄, dtype: object

In [82]:
c = (
    Pie(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .add(
        "",
        [list(z) for z in zip(data.年龄, data.七普)],
        radius=["30%", "55%"],
        center=["40%", "50%"]
    )
    .add(
        "",
        [list(z) for z in zip(data.年龄, data.六普)],
        center=["81%", "65%"],
        radius=["10%","20%"]
    )
    .set_global_opts(
        title_opts=opts.TitleOpts(title="年龄构成"),
        legend_opts=opts.LegendOpts(orient="vertical", pos_top="12%", pos_left="75%"),
    )
    .set_series_opts(label_opts=opts.LabelOpts(formatter="{b}: {c}"))
    #.render("pie_radius.html")
)
c.load_javascript()
c.render_notebook()
#make_snapshot(snapshot, c.render(), "年龄占比01.png")

### 对比2010年与2020年浙江省的年龄结构，60岁及以上人口比例提升了4.81%，而15~59岁人口比例降低了5.01%。

### 根据2021中国老龄化大数据报告，浙江省11个地市中已有6个地市深度老龄化。侧面反映出育龄群体正在缩小。

In [63]:
from pyecharts.charts import Bar

## 老龄人口

In [64]:
c = (
    Bar(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .add_xaxis(['55岁~59岁','60岁~64岁'])
    .add_yaxis("商家A", [486.05,350.63], category_gap="60%")

    .set_global_opts(xaxis_opts=opts.AxisOpts(axislabel_opts=opts.LabelOpts(rotate=-20)),)
    
    #.render("bar_border_radius.html")
)
c.load_javascript()
c.render_notebook()
# make_snapshot(snapshot, c.render(), "老龄人口.png")

### 除去省外人口，根据公安部数据，浙江户籍人口中60岁及以上的人口占比达到23.4%。

### 通过上述图表可知，未来五年内浙江省将有大量人口步入老龄化，五年后的浙江老龄化程度还将明显提升。

### 所以：育龄群体规模缩小，在未来缩小速率恐怕会越来越快。

# 4、经济压力

## 城乡居民社会养老保险基金

### 在分析育龄妇女人数时我们得知，浙江省老龄化程度大，那么我选择从养老金的层面来入手

In [70]:
c = (
    Bar(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .add_xaxis(['2013','2014','2015','2016','2017','2018','2019'])
    .add_yaxis("城乡居民社会养老保险基金收入", ['121.1','136.6','148.3','149.9','158.6','176.6','176.4'],gap="0%")
    .add_yaxis("城乡居民社会养老保险基金支出", ['105.3','126.6','143.3','143.4','157.5','171.5','178.9'],gap="0%")
    .set_series_opts(
        label_opts=opts.LabelOpts(is_show=False),
        markline_opts=opts.MarkLineOpts(
            data=[
                opts.MarkLineItem(y=178.9, name="yAxis=50"),
            ]
        ),
    )
    .set_global_opts(
        yaxis_opts=opts.AxisOpts(
            min_=100, 
#             max_=1.25,
        ),
    )

#     .render("bar_with_brush.html")
)
c.load_javascript()
c.render_notebook()
#make_snapshot(snapshot, bar.render(), "养老.png")

### 从上图可知，2019年浙江省城乡居民养老保险基金的支出已大于收入，虽然这只是其中一方面，但这无疑是对社会的一个巨大的负担。老龄化的加剧，劳动力的减少，浙江人的养老负担日益加重。

## 房价

In [66]:
house = pd.read_excel('房价6月.xlsx')
house

城市     房价
0   杭州  31368
1   宁波  25954
2   丽水  21846
3   台州  13935
4   湖州  11925
5   绍兴  16273
6   嘉兴  14115
7   舟山  17013
8   温州  17362
9   衢州  15107
10  金华  16778

In [67]:
c = (
    Bar(init_opts=opts.InitOpts(theme=ThemeType.VINTAGE))
    .add_xaxis(['杭州','宁波','丽水','台州','湖州','绍兴','嘉兴','舟山','温州','衢州','金华'])
    .add_yaxis("2020年各市人均可支配收入", ['61879','59952','37744','50643','54667','56600','51800','55830','54025','57935','50580'],stack="stack1")
    .add_yaxis("2021年7月购买100平方米房屋均价", ['3136800','2595400','2184600','1393500','1192500','1627300','1411500','1701300','1736200','1510700','1677800'],stack="stack1")
    .set_series_opts(
        label_opts=opts.LabelOpts(is_show=False),
    )
    .set_global_opts(
        yaxis_opts=opts.AxisOpts(
            min_=0, 
#             max_=1.25,
        ),
    )

#     .render("bar_with_brush.html")
)
#make_snapshot(snapshot, c.render(), "房价.png")
c.load_javascript()
c.render_notebook()

### 2016年全国房价飙升，浙江也不例外，其中代表杭州。杭州作为新一线城市房价高或许可以理解，但是作为二三线城市的丽水的房价竟也直逼杭州。

In [68]:
x = ['3136800','2595400','2184600','1393500','1192500','1627300','1411500','1701300','1736200','1510700','1677800']
y = ['61879','59952','37744','50643','54667','56600','51800','55830','54025','57935','50580']
z = []
for i in range (12):
    c = int(x[i])/int(y[i])
    print(c)

50.692480486109986
43.291299706431815
57.879398050021194
27.516142408625083
21.813891378711105
28.750883392226147
27.24903474903475
30.47286405158517
32.13697362332253
26.07577457495469
33.17121391854488


IndexError: list index out of range

### 如果我们用2021年7月份浙江省各市的平均房价与2020年各市的年人均可支配收入来进行粗略计算，杭州大约需要50.69年，需最长时间的丽水大约需要57.87年，而需要最短时间的湖州也要28.75年。当然这只是一个粗略计算，并没有考虑首付，税等等。但从这份简陋的数据中，仍能看出高昂的房价是压在年轻人身上另一座大山。

## 生育成本

### 生育成本是一个难以统计的数字，但有人计算出杭州抚养一个孩子到大学毕业，大约需要187万，全国排民第5。这是一个惊人的数字，而我根据自己在生活中的经验和网上对物价的评论，也能感受到物价的飞涨。但是工资却并没有涨多少。根据我浙江同学的描述，她们的教育竞争激烈，每年的补习费更是一笔不低的费用。鸡娃，从小做起。由此看来，生育成本是压在年轻人的第三座大山。

### 养老成本，房价，生育成本这三座压在年轻人身上的三座大山，压得年轻人喘不过气来。也是浙江省提高生育率的拦路虎。近年来引起热议的“内卷”“躺平”或许也是反应当下年轻人的心态。

# 未来

### 2021年推出了许多政策，浙江省建设共同富裕示范区，“三孩”政策正式实施，“双减”政策等等。有了二胎政策的前车之鉴，浙江省该如何响应政策，这对生育率又有何影响呢？